app/models/qualification.rb

```diff
@@ -32,10 +32,10 @@ def remove_inapplicable_data
   def display_attributes
     if secondary?
       %w[institution year]
+    elsif finished_studying?
+      %w[subject institution grade year]
     else
-      %w[subject institution].tap do |attrs|
-        attrs << %i[grade year] if finished_studying?
-      end
+      %w[subject institution]
```

Comment:
Subject is an optional field

Reply:
Hmm, this was there before though – I think I forgot a "don't show this if" somewhere else 😞